spec/javascripts/govuk_publishing_components/analytics-ga4/pii-remover.spec.js

```diff
@@ -0,0 +1,230 @@
+    })
+  })
+
+  describe('when configured to remove all PII', function () {
+    beforeEach(function () {
+      pageWantsDatesStripped()
+      pageWantsPostcodesStripped()
+      pii = new GOVUK.PIIRemover()
+    })
+
+    it('strips email addresses, postcodes and dates from strings', function () {
+      var string = pii.stripPII('this is an@email.com address, this is a sw1a 1aa postcode, this is a 2019-01-21 date')
+      expect(string).toEqual('this is [email] address, this is a [postcode] postcode, this is a [date] date')
+    })
+
+    it('strips all PII from objects', function () {
+      var obj = {
+        email: 'this is an@email.com address',
+        postcode: 'this is a sw1a 1aa postcode',
+        date: 'this is a 2019-01-21 date',
+        uuid: 'd6c2de5d-ef90-45d1-82d4-5f2438369eea'
+      }
+
+      var strippedObj = {
+        email: 'this is [email] address',
+        postcode: 'this is a [postcode] postcode',
+        date: 'this is a [date] date',
+        uuid: 'd6c2de5d-ef90-45d1-82d4-5f2438369eea'
+      }
+
+      obj = pii.stripPII(obj)
+      expect(obj).toEqual(strippedObj)
+    })
+
+    it('strips all PII from arrays', function () {
+      var arr = [
+        'this is an@email.com address',
+        'this is a sw1a 1aa postcode',
+        'this is a 2019-01-21 date'
+      ]
+
+      var strippedArr = [
+        'this is [email] address',
+        'this is a [postcode] postcode',
+        'this is a [date] date'
+      ]
+
+      arr = pii.stripPII(arr)
+      expect(arr).toEqual(strippedArr)
+    })
+  })
+
+  describe('when there is a a govuk:static-analytics:strip-postcodes meta tag present', function () {
+    beforeEach(function () {
+      pageWantsPostcodesStripped()
+      pii = new GOVUK.PIIRemover()
+    })
+
+    it('observes the meta tag and strips out postcodes', function () {
+      expect(pii.stripPostcodePII).toEqual(true)
+      var string = pii.stripPII('this is an@email.com address, this is a sw1a 1aa postcode, this is a long GL194LYX postcode, this is a 2019-01-21 date, this is a d6c2de5d-ef90-45d1-82d4-5f2438369eea content ID, this is a p800refund')
+      expect(string).toEqual('this is [email] address, this is a [postcode] postcode, this is a long [postcode] postcode, this is a 2019-01-21 date, this is a d6c2de5d-ef90-45d1-82d4-5f2438369eea content ID, this is a p800refund')
+    })
+
+    it('doesn\'t strip out UUIDs in URLs', function () {
+      expect(pii.stripPostcodePII).toEqual(true)
+      var string = pii.stripPII('gov.uk/thing?postcode=sw1a1aa&uuid=d6c2de5d-ef90-45d1-82d4-5f2438369eea')
+      expect(string).toEqual('gov.uk/thing?postcode=[postcode]&uuid=d6c2de5d-ef90-45d1-82d4-5f2438369eea')
+    })
+  })
+
+  describe('when there is a govuk:static-analytics:strip-query-string-parameters meta tag present', function () {
+    afterEach(function () {
+      resetHead()
```

Comment:
You don't need to call `resetHead` again here - any `afterEach` called at a higher level applies to any tests beneath it, so it's already been called by the main `afterEach`.